### PR TITLE
Issue/2228 m2 visibility

### LINKED ISSUE #2228: Edit Product Release M2: provide option to update product visibility
As part of Product Settings, we want to allow a user to edit the product's visibility.

Tasks (feel free to add new subtasks):
- [ ] Add the new cell "Visibility" in Product Settings
- [ ] Add the new screen "Visibility"

## Screenshots
| Visibility screen             |  Product Settings |
:-------------------------:|:-------------------------:
Schermata 2020-04-09 alle 15 30 44 | Schermata 2020-04-09 alle 15 30 54

### PR DESCRIPTION
**Pull request by nbradbury**
Closes #2228 - Adds product visibility to product settings. This one is a little tricky because "visibility" and "password" aren't product fields. 

The visibility is determined by the status and the password. If the password isn't empty, then visibility is `PASSWORD_PROTECTED`. If there's no password and the product status is `PRIVATE` then the visibility is `PRIVATE`, otherwise it's `PUBLIC`.

To get the password, we have to make a separate request to the WP.com endpoint `/sites/$site/posts/$post_ID`. This adds some complexity because until we know whether there's a password we can't determine the product's visibility.

Screenshot_1587761557

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductStatus.kt [hunk 1]
```diff
@@ -15,12 +15,12 @@ enum class ProductStatus(@StringRes val stringResource: Int = 0, val value: Stri
     PENDING(R.string.product_status_pending, CoreProductStatus.PENDING.value),
     PRIVATE(R.string.product_status_private, CoreProductStatus.PRIVATE.value);
 
-    fun toLocalizedString(context: Context): String {
+    fun toLocalizedString(context: Context, long: Boolean = false): String {
```
Comment (anitaa1990) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductStatus.kt:18]: np: I might be missing some context here but not really sure what `long` represents here. Can we maybe add a comment on what it means or change the name to something more clear?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/settings/ProductVisibilityFragment.kt [hunk 2]
```diff
@@ -41,70 +40,101 @@ class ProductVisibilityFragment : BaseProductSettingsFragment(), OnClickListener
...
+            showPassword(it == btnPasswordProtected)
+        }
+    }
+
+    private fun showPassword(show: Boolean) {
+        if (show && editPassword.visibility != View.VISIBLE) {
+            editPassword.visibility = View.VISIBLE
+            editPassword.requestFocus()
+            ActivityUtils.showKeyboard(editPassword)
+        } else if (!show && editPassword.visibility == View.VISIBLE) {
```
Comment (anitaa1990) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/settings/ProductVisibilityFragment.kt:86]: This is not really related to this PR but I feel like we could maybe create a `ViewModel` class for the `BaseProductSettingsFragment` just so we can migrate the business logic to the ViewModel and unit test it. This is more of an observation really so feel free to make changes when needed :)

**Review by anitaa1990 — COMMENTED** [review]
Changes look good @nbradbury! I just left a small nitpick in the code and a suggestion! Ping me once the nit is addressed and we can :shipit: Nice Work!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductStatus.kt (hunk 1 above)
Comment (nbradbury) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductStatus.kt:23]: > I might be missing some context here but not really sure what `long` represents here. Can we maybe add a comment on what it means or change the name to something more clear?

Good call, I added a comment in f7a4098. This is really only about private status. In product detail and the product list we only want to show "Private," but in the status screen we want to show "Privately published" to match the web UI.

Comment (anitaa1990) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductStatus.kt:23]: Thanks!

**Review by anitaa1990 — APPROVED** [review]
Looks good :shipit:

**Pull request merged**